pykern/pkcli/ci.py

```diff
@@ -120,15 +120,27 @@ def _error(m):
 
     d = _CHECK_FILES[case]
     r = []
-    p = pkio.py_path()
     n = 0
-    for f in pkio.walk_tree(p, d.include_files):
-        f = p.bestrelpath(f)
-        if re.search(d.exclude_files, f):
-            continue
-        n += 1
-        for l in check_file(pkio.read_text(f).split("\n")):
-            r.append(f"{f}{l}")
+
+    def _paths():
+        if pkio.py_path("setup.py").isfile() and (
+            pkio.py_path("README.rst").isfile() or pkio.py_path("README.md").isfile()
+        ):
+            return (
+                pkio.py_path().basename,
+                "tests",
+                "setup.py",
+            )
+        return (pkio.py_path(),)
+
+    for p in _paths():
+        for f in pkio.walk_tree(p, d.include_files):
+            f = pkio.py_path().bestrelpath(f)
```

Comment:
pkio.py_path() is a loop invariant. Need to pull that out of the loop, because getting cwd is expensive relative to the operations.

Reply:
2793e544f3899eefb18ae9d4c895cdffdc1483c8